pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -162,6 +162,12 @@ func (s *Service) ReconcileSecurityGroups() error {
 
 func (s *Service) DeleteSecurityGroups() error {
 	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
+
+	if s.scope.VPC().ID == "" {
+		conditions.MarkFalse(s.scope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
+		return nil
+	}
+
 	if err := s.scope.PatchObject(); err != nil {
```

Comment:
Just noticed that we are not patching the conditions on early return. We can move this to a defer func above.

Reply:
This is not true as well. It is being patched on close.